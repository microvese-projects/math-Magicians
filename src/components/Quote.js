import { useEffect, useState } from 'react';

export default function Quote() {
  const [error, setError] = useState(null);
  const [loading, setLoading] = useState(false);
  const [data, setData] = useState('');

  useEffect(() => {
    setLoading(true);
    const fetchQoute = async () => {
      try {
        const res = await fetch('https://api.api-ninjas.com/v1/quotes?category=happiness', {
          method: 'GET',
          withCredentials: true,
          headers: {
            'X-Api-Key': 'JPH0dr8vp5cPyFyCj/jr7g==7Z5eQyE7zjkxZ4gW',
            'Content-Type': 'application/json',
          },
        });
        const json = await res.json();
        setData(json[0]);
      } catch (err) {
        setError(err);
      }
      setLoading(false);
    };

    fetchQoute();
  }, [setData, setLoading, setError]);

  if (error) return <div className="error">Something went wrong!! Try Again.</div>;

  if (loading) return <div className="loading-message">Loading...</div>;

  return (
    <div className="quote-container">
      <p className="quote">{data.quote}</p>
      <p className="author">{data.author}</p>
    </div>
  );
}
