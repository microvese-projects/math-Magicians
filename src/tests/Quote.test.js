import React from 'react';
import renderer from 'react-test-renderer';
import Quote from '../components/Quote';

test('Should render calculator correctly', () => {
  const tree = renderer
    .create(<Quote />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
