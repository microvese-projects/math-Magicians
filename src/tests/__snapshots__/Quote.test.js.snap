// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render calculator correctly 1`] = `
<div
  className="quote-container"
>
  <p
    className="quote"
  />
  <p
    className="author"
  />
</div>
`;
